tests/integration_tests/style/test_gitlint.py

```diff
@@ -14,6 +15,23 @@ def test_gitlint():
     """
     os.environ['LC_ALL'] = 'C.UTF-8'
     os.environ['LANG'] = 'C.UTF-8'
+    # devtool script will set the working directory to FC_ROOT/tests.
+    # We will need the parent for safe.directory configuration
+    working_dir = Path(os.getcwd())
+
+    # The firecracker root git folder in the container is
+    # bind-mounted to a folder on the host which is mapped to a
+    # user that is different from the user which runs the integ tests.
```

Comment:
I think it's possible to coax Docker to define permissions/users that unblock git. However some operations are privileged which results in root ownership of an entity in the guest. This would/should always misalign with the host user (I would never expect the development environment to be operating as root). If we really wanted to pursue this route, we should be introducing a non-root privileged user in the guest named after the host user, but I feel this is out of scope and has limited value in my opinion.